Examples/Io/Root/src/RootTrajectoryParametersReader.cpp

```diff
@@ -0,0 +1,187 @@
+}
+
+std::string ActsExamples::RootTrajectoryParametersReader::name() const {
+  return m_cfg.name;
+}
+
+std::pair<size_t, size_t>
+ActsExamples::RootTrajectoryParametersReader::availableEvents() const {
+  return {0u, m_events};
+}
+
+ActsExamples::RootTrajectoryParametersReader::
+    ~RootTrajectoryParametersReader() {
+  delete m_multiTrajNr;
+  delete m_subTrajNr;
+  delete m_t_barcode;
+  delete m_t_charge;
+  delete m_t_time;
+  delete m_t_vx;
+  delete m_t_vy;
+  delete m_t_vz;
+  delete m_t_px;
+  delete m_t_py;
+  delete m_t_pz;
+  delete m_t_theta;
+  delete m_t_phi;
+  delete m_t_pT;
+  delete m_t_eta;
+  delete m_hasFittedParams;
+  delete m_eLOC0_fit;
+  delete m_eLOC1_fit;
+  delete m_ePHI_fit;
+  delete m_eTHETA_fit;
+  delete m_eQOP_fit;
+  delete m_eT_fit;
+  delete m_err_eLOC0_fit;
+  delete m_err_eLOC1_fit;
+  delete m_err_ePHI_fit;
+  delete m_err_eTHETA_fit;
+  delete m_err_eQOP_fit;
+  delete m_err_eT_fit;
+}
+
+ActsExamples::ProcessCode ActsExamples::RootTrajectoryParametersReader::read(
+    const ActsExamples::AlgorithmContext& context) {
+  ACTS_DEBUG("Trying to read recorded tracks.");
+
+  // read in the fitted track parameters and particles
+  if (m_inputChain && context.eventNumber < m_events) {
+    // lock the mutex
+    std::lock_guard<std::mutex> lock(m_read_mutex);
+    // now read
+
+    std::shared_ptr<Acts::PerigeeSurface> perigeeSurface =
+        Acts::Surface::makeShared<Acts::PerigeeSurface>(
+            Acts::Vector3(0., 0., 0.));
+
+    // The collection to be written
+    std::vector<Acts::BoundTrackParameters> trackParameterCollection;
+    SimParticleContainer truthParticleCollection;
+
+    // Read the correct entry: batch size * event_number + ib
+    m_inputChain->GetEntry(context.eventNumber);
+    ACTS_VERBOSE("Reading entry: " << context.eventNumber);
+
+    unsigned int nTracks = m_eLOC0_fit->size();
+    for (unsigned int i = 0; i < nTracks; i++) {
+      Acts::BoundVector paramVec;
+      paramVec << (*m_eLOC0_fit)[i], (*m_eLOC1_fit)[i], (*m_ePHI_fit)[i],
+          (*m_eTHETA_fit)[i], (*m_eQOP_fit)[i], (*m_eT_fit)[i];
+
+      // Resolutions
+      double resD0 = (*m_err_eLOC0_fit)[i];
+      double resZ0 = (*m_err_eLOC1_fit)[i];
+      double resPh = (*m_err_ePHI_fit)[i];
+      double resTh = (*m_err_eTHETA_fit)[i];
+      double resQp = (*m_err_eQOP_fit)[i];
+      double resT = (*m_err_eT_fit)[i];
+
+      // Fill vector of track objects with simple covariance matrix
+      Acts::BoundSymMatrix covMat;
+
+      covMat << resD0 * resD0, 0., 0., 0., 0., 0., 0., resZ0 * resZ0, 0., 0.,
+          0., 0., 0., 0., resPh * resPh, 0., 0., 0., 0., 0., 0., resTh * resTh,
+          0., 0., 0., 0., 0., 0., resQp * resQp, 0., 0., 0., 0., 0., 0.,
+          resT * resT;
+
+      trackParameterCollection.push_back(Acts::BoundTrackParameters(
+          perigeeSurface, paramVec, std::move(covMat)));
+    }
+
+    unsigned int nTruthParticles = m_t_vx->size();
```

Comment:
So the `nTruthParticles` should be equal to the `nTracks`, right? You might want to add an ERROR in case this is not the case.